src/pages/en/core-concepts/framework-components.md

```diff
@@ -0,0 +1,155 @@
+---
+layout: ~/layouts/MainLayout.astro
+title: Framework Components
+description: Learn how to use React, Svelte, etc.
+---
+Astro allows you to build your site using JavaScript UI components from a variety of popular frameworks. Astro will run your [React](https://reactjs.org/), [Preact](https://preactjs.com/), [Svelte](https://svelte.dev/), [Vue](https://vuejs.org/), [SolidJS](https://www.solidjs.com/), [AlpineJS](https://alpinejs.dev/) and [Lit](https://lit.dev/) components at build time, then render your entire site to static HTML by default, removing all the JavaScript from the final page!
+
+```astro
+---
+import MyReactComponent from '../components/MyReactComponent.jsx';
+---
+<html>
+  <head>
+    <title>Look How Easy it is to Use a React Component!</title>
+  </head>
+  <body>
+    <h1>My Astro Page</h1>
+    <MyReactComponent />
+  </body>
+</html>
+```
+
+You can use any of these frameworks as templating languages to build static pages, taking advantage of the syntax and tooling you already know and love. This allows you to build with components you're familiar with, maybe even the exact same ones you've already written! And, you won't send any unnecessary JavaScript to the client to render static elements, even dynamically generated ones.
```

Comment:
can you do an audit for unnecessary `!`s? I think the one here after `written` is unnecessary, and we should save them for the actually exciting stuff :)

Reply:
also, this sentence is a bit tough to parse, can we simplify?: `And, you won't send any unnecessary JavaScript to the client to render static elements, even dynamically generated ones.`